### ISSUE TITLE
[Web] Unable to change log level and hide warning messages when creating inference session from buffer

### ISSUE THREAD

**Issue by xenova** [microsoft/onnxruntime#17377]
### Describe the issue

It is impossible to disable warning messages generated when creating an onnx session if you use a buffer to create the model (instead of a path).

Linked to:
- https://github.com/microsoft/onnxruntime/issues/14694

cc @fs-eire @guschmue 

### To reproduce

1. Get model files. For this example, I'll use [whisper-tiny.en](https://huggingface.co/Xenova/whisper-tiny.en)'s [decoder](https://huggingface.co/Xenova/whisper-tiny.en/resolve/main/onnx/decoder_model_merged_quantized.onnx), but it happens for most models.
    ```sh
    git clone https://huggingface.co/Xenova/whisper-tiny.en
    ```
    or just download the single onnx file needed from [here](https://huggingface.co/Xenova/whisper-tiny.en/resolve/main/onnx/decoder_model_merged_quantized.onnx).
2. Run this code:
    ```js
    
    import fs from 'fs';
    import ort from 'onnxruntime-node'
    
    const options = {
        executionProviders: ['cpu'],
        graphOptimizationLevel: 'disabled',
    }
    
    const path = './models/whisper-tiny.en/onnx/decoder_model_merged_quantized.onnx';
    const session = await ort.InferenceSession.create(path, options);
    console.log(session);
    
    // Read as buffer
    const buffer = fs.readFileSync(path);
    const sessionBroken = await ort.InferenceSession.create(buffer, options);
    console.log(sessionBroken);
    ```

3. See output:
    WORKING:
    ```
    m {
      handler: OnnxruntimeSessionHandler {
        inputNames: [...],
        outputNames: [...]
      }
    }
    ```

    BROKEN:
    ```
    2023-08-31 21:25:16.663056737 [W:onnxruntime:, graph.cc:3490 CleanUnusedInitializersAndNodeArgs] Removing initializer '/model/decoder/Shape_4_output_0'. It is not used by any node and should be removed from the model.
    ...
    2023-08-31 21:25:16.718106598 [W:onnxruntime:, graph.cc:3490 CleanUnusedInitializersAndNodeArgs] Removing initializer '/model/decoder/layers.1/final_layer_norm/Constant_1_output_0'. It is not used by any node and should be removed from the model.
    m {
      handler: OnnxruntimeSessionHandler {
        inputNames: [...],
        outputNames: [...]
      }
    }
    ```

### Urgency

Blocks the following issues from being fixed in transformers.js:
- https://github.com/xenova/transformers.js/issues/117
- https://github.com/xenova/transformers.js/issues/270

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.15.1

### Execution Provider

'wasm'/'cpu' (WebAssembly CPU)

**Comment by hariharans29** [comment]
CC @fs-eire @guschmue

**Comment by fs-eire** [comment]
I build main branch and tested locally and cannot reproduce this issue. I will do a test on 1.15.1 later

**Comment by fs-eire** [comment]
I can reproduce this issue on 1.15.1 and 1.16.0. However, my local build seems has no issue. I am taking further investigations.

**Comment by fs-eire** [comment]
I found the root cause. 

So this was a bug and it's [fixed](https://github.com/microsoft/onnxruntime/commit/c26287921456c0f99665c3d9c85eb0ab0b5347c7#diff-bfbe3368d88d623d9211f7cf5ea963757216be3515c5edb420d0e68e067d0100R82) in main branch. However unfortunately v1.16 release does not include this change.

I will check if we are able to do this bugfix in 1.16.1 patch

**Comment by xenova** [comment]
Great, thanks for that!

**Comment by fs-eire** [comment]
v1.16.1 is released with this fix.

**Comment by wesbos** [comment]
Getting these again. Any way to filter them out? Its overwhelming my console

```
2024-03-01 16:26:38.495 node[30406:15443260] 2024-03-01 16:26:38.495795 [W:onnxruntime:, graph.cc:3490 CleanUnusedInitializersAndNodeArgs] Removing initializer '/model/decoder/layers.2/final_layer_norm/Constant_1_output_0'. It is not used by any node and should be removed from the model.
```

**Comment by fs-eire** [comment]
@wesbos The issue should be fixed in v1.16.1.

If you are just asking about how to deal with the warning messages, here are the solutions -
1. optimize the model: try to use the latest exporter to re-export the model, or use a model optimizer to optimize the model to remove the unused initializer. This is the way to solve the root cause.
2. set [logSeverityLevel](https://onnxruntime.ai/docs/api/js/interfaces/InferenceSession.SessionOptions.html#logSeverityLevel) in session options to 3 or above to disallow the message being print out
3. set [graphOptimizationLevel](https://onnxruntime.ai/docs/api/js/interfaces/InferenceSession.SessionOptions.html#graphOptimizationLevel) in session option to 'disabled' so that graph optimizer will not run. not recommended as this may slow down the model inferencing.

**Comment by AmitDJagtap** [comment]
For those who are still stumped, adding a nodejs code snippet here to set log level to 3 . 

warning levels
  ```
  VERBOSE = 0,
  INFO = 1,
  WARNING = 2,
  ERROR = 3,
  FATAL = 4
```
your code 
```

import { pipeline, env } from '@xenova/transformers';
import fs from 'fs';
env.cacheDir = './.cache';
env.backends.onnx.logSeverityLevel = 3; // this line here
```
